modules/@apostrophecms/rich-text-widget/index.js

```diff
@@ -205,7 +212,38 @@ module.exports = {
         return widget.content;
       },
 
+      // Handle permalinks
       async load(req, widgets) {
+        const widgetsByDocId = new Map();
+        let ids = [];
+        const project = self.options.project;
+        for (const widget of widgets) {
+          if (!widget.permalinkIds) {
+            return;
+          }
+          for (const id of widget.permalinkIds) {
+            const docWidgets = widgetsByDocId.get(id) || [];
+            docWidgets.push(widget);
+            widgetsByDocId.set(id, docWidgets);
+            ids.push(id);
+          }
+        }
+        ids = [ ...new Set(ids) ];
+        if (!ids.length) {
+          return;
+        }
+        const docs = await self.apos.doc.find(req, {
+          aposDocId: {
+            $in: ids
+          }
+        }).project(project).toArray();
```

Comment:
```suggestion
        }).project(self.options.project).toArray();
```

to avoid declaring the variable above (I find it more readable this way)